### ISSUE TITLE
Auto approve and merge pull requests by Dependabot 

### ISSUE THREAD

**Issue by sophia-guo** [AdoptOpenJDK/openjdk-tests#2343]
Details see https://github.com/AdoptOpenJDK/run-aqa/issues/72. The requirement mainly fits to projects with typescript.

**Comment by sophia-guo** [comment]
Same requirement for openjdk-test-tools repo. https://github.com/AdoptOpenJDK/openjdk-test-tools/pull/372

**Comment by llxia** [comment]
We need to be very careful with the auto-merge. Some 3rd party lib upgrades will break the code. And we should not auto-merge any PR from a bot.

**Comment by sophia-guo** [comment]
Yes, we need to be careful about the auto merge. However it depends how we use it. `Dependabot will resolve any conflicts with its created PR as long as you don't alter it yourself. You can also trigger a rebase manually by commenting @dependabot rebase`. Also note: Dependabot will wait until all your status checks pass before merging. This is a function of Dependabot itself, https://github.com/marketplace/actions/dependabot-auto-merge. We could setup workflows for PRs created by Dependabot ( or could just setup the normal workflow it depends) so auto merge would not work if any status checks fail. Hence no build will be broken.

**Comment by llxia** [comment]
Talked with Sophia, we may need to evaluate this case by case. For https://github.com/AdoptOpenJDK/openjdk-test-tools, I think we need to create PR builds first.

**Comment by sophia-guo** [comment]
May also extend to PR of github-actions https://github.com/AdoptOpenJDK/openjdk-tests/pull/2319
